frontend/src/api/k8s/templates.ts

```diff
@@ -39,18 +46,43 @@ export const assembleServingRuntimeTemplate = (
   };
 };
 
-export const listTemplates = async (
-  namespace?: string,
-  labelSelector?: string,
-): Promise<TemplateKind[]> => {
-  const queryOptions = {
-    ...(namespace && { ns: namespace }),
-    ...(labelSelector && { queryParams: { labelSelector } }),
-  };
-  return k8sListResource<TemplateKind>({
-    model: TemplateModel,
-    queryOptions,
-  }).then((listResource) => listResource.items);
+export const useTemplates = (namespace?: string): CustomWatchK8sResult<TemplateKind[]> => {
+  const modelServingEnabled = useModelServingEnabled();
+  const customServingRuntimesEnabled = useCustomServingRuntimesEnabled();
+
+  const initResource: WatchK8sResource | null =
+    namespace && modelServingEnabled
+      ? {
+          isList: true,
+          groupVersionKind: groupVersionKind(TemplateModel),
+          namespace,
+        }
+      : null;
+
+  const [templatesData, loaded, error] = useCustomK8sWatchResource<TemplateKind[]>(
+    initResource,
+    TemplateModel,
+  );
+
+  const templates = React.useMemo(
+    () =>
+      customServingRuntimesEnabled
+        ? templatesData
+        : templatesData.filter(
+            (template) => template.metadata.labels?.['opendatahub.io/ootb'] === 'true',
+          ),
+    [templatesData, customServingRuntimesEnabled],
+  );
+
+  if (!namespace) {
+    return [[], true, new NotReadyError('No namespace provided')];
+  }
+
+  if (!modelServingEnabled) {
+    return [[], true, new NotReadyError('Model serving is not enabled')];
+  }
```

Comment:
I forgot that `NotReadyError` is a special error handled by `useFetchState` as a way to delay any response. When `useFetchState` encounters this error, it does not set `loaded` and instead returns: `[<default value>, loaded = false, error = undefined]`

I believe this would be the equivalent behavior to the old `useTemplates`. 
Note that we need to make sure that `templatesData` is an array because the old implementation defaults to empty array. We also cannot simply return `[]` here because that would be an unstable reference. Either `templatesData` needs to ensure it is always a value so that `useMemo` will make it stable or you need to define an empty array constant.

It may be possible to also return `[undefined, false, undefined]` since data shouldn't be looked at unless `loaded === true` but it could pose other problems.

```suggestion
  if (!namespace || !modelServingEnabled) {
    return [templatesData, false, undefined];
  }
```